src/ControllerMessenger.test.ts

```diff
@@ -289,3 +289,396 @@ describe('ControllerMessenger', () => {
+
+    expect(handler.callCount).toEqual(0);
+  });
+
+  it('should throw when unsubscribing when there are no subscriptions', () => {
+    type MessageEvent = { type: 'MessageController:message'; payload: [string] };
+    const controllerMessenger = new ControllerMessenger<never, MessageEvent>();
+    const restrictedControllerMessenger = controllerMessenger.getRestricted({
+      name: 'MessageController',
+      allowedActions: [],
+      allowedEvents: ['MessageController:message'],
+    });
+
+    const handler = sinon.stub();
+    expect(() => restrictedControllerMessenger.unsubscribe('MessageController:message', handler)).toThrow(
+      `Subscription not found for event: 'MessageController:message'`,
+    );
+  });
+
+  it('should throw when unsubscribing a handler that is not subscribed', () => {
+    type MessageEvent = { type: 'MessageController:message'; payload: [string] };
+    const controllerMessenger = new ControllerMessenger<never, MessageEvent>();
+    const restrictedControllerMessenger = controllerMessenger.getRestricted({
+      name: 'MessageController',
+      allowedActions: [],
+      allowedEvents: ['MessageController:message'],
+    });
+
+    const handler1 = sinon.stub();
+    const handler2 = sinon.stub();
+    restrictedControllerMessenger.subscribe('MessageController:message', handler1);
+
+    expect(() => restrictedControllerMessenger.unsubscribe('MessageController:message', handler2)).toThrow(
+      `Subscription not found for event: 'MessageController:message'`,
+    );
+  });
+
+  it('should not call subscriber after clearing event subscriptions', () => {
+    type MessageEvent = { type: 'MessageController:message'; payload: [string] };
+    const controllerMessenger = new ControllerMessenger<never, MessageEvent>();
+    const restrictedControllerMessenger = controllerMessenger.getRestricted({
+      name: 'MessageController',
+      allowedActions: [],
+      allowedEvents: ['MessageController:message'],
+    });
+
+    const handler = sinon.stub();
+    restrictedControllerMessenger.subscribe('MessageController:message', handler);
+    restrictedControllerMessenger.clearEventSubscriptions('MessageController:message');
+    restrictedControllerMessenger.publish('MessageController:message', 'hello');
+
+    expect(handler.callCount).toEqual(0);
+  });
+
+  it('should not throw when clearing event that has no subscriptions', () => {
+    type MessageEvent = { type: 'MessageController:message'; payload: [string] };
+    const controllerMessenger = new ControllerMessenger<never, MessageEvent>();
+    const restrictedControllerMessenger = controllerMessenger.getRestricted({
+      name: 'MessageController',
+      allowedActions: [],
+      allowedEvents: ['MessageController:message'],
+    });
+
+    expect(() => restrictedControllerMessenger.clearEventSubscriptions('MessageController:message')).not.toThrow();
+  });
+
+  it('should allow calling an external action', () => {
```

Comment:
nit: It _might_ be worth adding test cases that mix external and internal (relative to the namespace) actions and events.

For instance, we could copy the below case and modify it to have `allowedActions: ['CountController:count', 'OtherController:foo']`. Not necessary with respect to the implementation, just to cover an input edge case.

Reply:
Done in deb2409d48aa7d966eaa241a408d9afd5522731d